src/components/MetaMaskButton/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { Web3Provider } from '@ethersproject/providers';
+import { UnsupportedChainIdError, useWeb3React as useWeb3ReactCore } from '@web3-react/core';
+import {
+    InjectedConnector,
+    NoEthereumProviderError,
+    UserRejectedRequestError as UserRejectedRequestErrorInjected,
+  } from '@web3-react/injected-connector';
+import * as React from 'react';
+
+import { useDispatch } from 'react-redux';
+import { MetaMaskLogo } from '../../assets/images/MetaMaskLogo';
+import { Web3ProviderWrapper } from '../../helpers';
+import { alertPush } from '../../modules';
+
+interface OwnProps {
+    depositAddress: string;
+}
+
+type Props = OwnProps;
+
+const getErrorMessage = (error: Error): string => {
+    if (error instanceof NoEthereumProviderError) {
+        return 'metamask.error.noExtension';
+    } else if (error instanceof UnsupportedChainIdError) {
+        return 'metamask.error.unsupportedNetwork';
+    } else if (error instanceof UserRejectedRequestErrorInjected) {
+        return 'metamask.error.unauthorized';
+    } else {
+        window.console.error(error);
+
+        return 'metamask.error.unknown';
+    }
+};
```

Comment:
move it to the helper

Reply:
no need, it's using only here and won't be used by any other component in future

Reply:
Then move this logic to the handleErrorSaga